src/coord/src/coord.rs

```diff
@@ -2861,34 +2848,70 @@ impl Coordinator {
     /// schemas with the same timeline as whatever the first query is".
     fn timedomain_for(
         &self,
-        source_ids: &[GlobalId],
-        source_timeline: &Option<Timeline>,
+        uses_ids: &[GlobalId],
+        timeline: &Option<Timeline>,
         conn_id: u32,
-    ) -> Result<Vec<GlobalId>, CoordError> {
-        let mut timedomain_ids = self
-            .catalog
-            .schema_adjacent_indexed_relations(&source_ids, conn_id);
-
-        // Filter out ids from different timelines. The timeline code only verifies
-        // that the SELECT doesn't cross timelines. The schema-adjacent code looks
-        // for other ids in the same database schema.
-        timedomain_ids.retain(|&id| {
-            let id_timeline = self
-                .validate_timeline(vec![id])
-                .expect("single id should never fail");
-            match (&id_timeline, &source_timeline) {
-                // If this id doesn't have a timeline, we can keep it.
-                (None, _) => true,
-                // If there's no source timeline, we have the option to opt into a timeline,
-                // so optimistically choose epoch ms. This is useful when the first query in a
-                // transaction is on a static view.
-                (Some(id_timeline), None) => id_timeline == &Timeline::EpochMilliseconds,
-                // Otherwise check if timelines are the same.
-                (Some(id_timeline), Some(source_timeline)) => id_timeline == source_timeline,
-            }
-        });
+    ) -> Result<(Vec<GlobalId>, Vec<GlobalId>), CoordError> {
```

Comment:
Maybe update the doccomment to explain which of these are storage v compute ids?

Reply:
It's especially awful because we're inconsistent: some methods go `(storage, compute)` and some go `(compute, storage)`. If CI suggests this PR isn't a total tire fire, I might introduce a type like

```rust
struct IdBundle {
    storage_ids: Vec<GlobalId>,
    compute_ids: Vec<GlobalId>,
}
```

so that it's relatively impossible to mix it up. Thoughts? Naming suggestions?

Reply:
Seems good! I usually go `storage` then `compute` because that's how I think of the layers, but 100% it's clearer with that struct. I .. wonder if it will always want a `ComputeInstanceId` with it too? I'm not sure. Maybe it changes with the SET CLUSTER work.

Reply:
> Seems good! I usually go `storage` then `compute` because that's how I think of the layers, but 100% it's clearer with that struct. I .. wonder if it will always want a `ComputeInstanceId` with it too? I'm not sure. Maybe it changes with the SET CLUSTER work.

Yeah, that's what I tried to stick to in all code that didn't already have the opposite opinion (looking at you, `nearest_indexes`), but homogenous tuples without an obvious ordering on the fields stress me out!

I think `nearest_indexes`, `determine_timestamp`, and `timedomain_for` all want to operate on an `IdBundle` without a compute instance ID for now, but possibly that will change in the future!